drivers/temperature/adt7420/adt7420.c

```diff
@@ -40,55 +40,107 @@
 /******************************************************************************/
 /***************************** Include Files **********************************/
 /******************************************************************************/
+#include <stdint.h>
 #include <stdlib.h>
+#include "spi.h"
+#include "i2c.h"
+#include "error.h"
 #include "adt7420.h"
 
+const struct adt7420_chip_info chip_info[] = {
+	[ID_ADT7410] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7420] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7422] = {
+		.resolution = 16,
+		.communication = I2C,
+	},
+	[ID_ADT7310] = {
+		.resolution = 16,
+		.communication = SPI,
+	},
+	[ID_ADT7320] = {
+		.resolution = 16,
+		.communication = SPI,
+	}
+};
+
 /***************************************************************************//**
- * @brief Reads the value of a register.
+ * @brief Reads the value of a register SPI/I2C interface.
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
  *
  * @return register_value  - Value of the register.
 *******************************************************************************/
-uint8_t adt7420_get_register_value(struct adt7420_dev *dev,
-				   uint8_t register_address)
+uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
+				    uint8_t register_address)
 {
 	uint8_t register_value = 0;
+	uint8_t data[2] = {0,0xFF};
+
+	if (chip_info[dev->active_device].communication == SPI) {
+		/* Form the SPI register read command byte. Bits [5:3] of command byte
```

Comment:
What about creating 2 function instead of this if/else:
adt7420_i2c_get_register_value and adt7420_spi_get_register_value.

Reply:
Two new functions:
adt7420_i2c_get_register_value () and
adt7420_spi_get_register_value() 
has been created in place of adt7420_get_register_value()